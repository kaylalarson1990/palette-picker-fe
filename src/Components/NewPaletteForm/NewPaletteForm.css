.toggle-form {
  padding: 10px;
  background-color: white;
  border: 1px black solid;
  color: black;
  font-size: 18px;
  border-radius: 5px;
  margin-top: 10px;
}

.show-form {
  display: flex;
  flex-direction: column;
  align-items: center;
}

select {
  margin: 10px;
  width: 300px;
  padding: 15px;
}

input {
  width: 300px;
  padding: 8px;
  border-radius: 5px;
}
