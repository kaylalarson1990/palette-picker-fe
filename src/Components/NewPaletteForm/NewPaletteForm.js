import React, {Component} from 'react';
import './NewPaletteForm.css'

export class NewPaletteForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      form: false,
      palatteName: ''
    }
  }

  populateForm = () => {
    this.setState({ form: true})
  }

  handleChange = e => {
    this.setState({
      palatteName: e.target.value
    })
  }

  handleSubmit = e => {
    e.preventDefault();
    this.setState({ form: false})
  }

  render() {
    return (
      <section>
      {!this.state.form && <button className='toggle-form' onClick={e => this.populateForm()}>Save palette</button>}
      { this.state.form && 
        <form>
          <input className='palette-name' placeholder='Enter Palatte Name...' value={this.state.palatteName} onChange={e => this.handleChange(e)}/>
          <button onClick={e => this.handleSubmit(e)}>Submit</button>
        </form>
      }
      </section>
    )
  }
}

export default NewPaletteForm;