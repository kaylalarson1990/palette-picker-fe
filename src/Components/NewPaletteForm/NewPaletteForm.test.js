import React from "react";
import { shallow } from "enzyme";
import { NewPaletteForm, mapStateToProps } from "./NewPaletteForm";
import "../../../setupTests";

describe("NewPaletteForm", () => {
  let wrapper, instance, props;
  beforeEach(() => {
    props = {
      projects: [{ project_name: "Test Project" }]
    };
    wrapper = shallow(<NewPaletteForm {...props} />);
    instance = wrapper.instance();
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it.skip("should be called on input change", () => {
    const mockEvent = {
      target: { name: "palette-name", value: "Test Palette" }
    };
    instance.handleChange = jest.fn();
    wrapper.find(".palette-input-name").simulate("change", mockEvent);
    expect(wrapper.state("palette_name")).toEqual("Test Palette");
  });

  it.skip("should call handleSubmit when save button is clicked", () => {
    const mockEvent = { preventDefault: jest.fn() };
    instance.handleSubmit = jest.fn();

    wrapper.find(".palette-submit").simulate("click", mockEvent);
    expect(instance.handleSubmit).toHaveBeenCalledWith(mockEvent);
  });

  it.skip("should post a new project when handleSubmit is called", () => {
    const mockName = "Dream House";
    instance.postProject = jest.fn();

    instance.postProject(mockName);
    expect(instance.postProject).toHaveBeenCalledWith(mockName);
  });
});

describe("mapStateToProps", () => {
  it("should return an object with a palettes array", () => {
    const mockState = {
      palettes: []
    };
    const expected = {
      palettes: []
    };
    const mappedProps = mapStateToProps(mockState);
    expect(mappedProps).toEqual(expected);
  });
});
