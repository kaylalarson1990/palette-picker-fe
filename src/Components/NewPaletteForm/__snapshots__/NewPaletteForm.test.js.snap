// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewPaletteForm should match the snapshot 1`] = `
<section>
  <button
    className="toggle-form toggle-btn"
    onClick={[Function]}
  >
    Save a palette
  </button>
</section>
`;
